src/rpc_server.rs

```diff
@@ -238,6 +253,50 @@ impl AtlasPriorityFeeEstimatorRpcServer for AtlasPriorityFeeEstimator {
         &self,
         get_priority_fee_estimate_request: GetPriorityFeeEstimateRequest,
     ) -> RpcResult<GetPriorityFeeEstimateResponse> {
+        self.execute_priority_fee_estimate_coordinator(get_priority_fee_estimate_request,  move |
+            accounts: Vec<Pubkey>,
+            include_vote: bool,
+            lookback_period: Option<u32>| {
+             self.priority_fee_tracker.get_priority_fee_estimates(accounts, include_vote, lookback_period)
+        })
+
+    }
+
+    fn get_priority_fee_estimate2(
+        &self,
+        get_priority_fee_estimate_request: GetPriorityFeeEstimateRequest,
+    ) -> RpcResult<GetPriorityFeeEstimateResponse> {
+        self.execute_priority_fee_estimate_coordinator(get_priority_fee_estimate_request,  |
+            accounts: Vec<Pubkey>,
+            include_vote: bool,
+            lookback_period: Option<u32>| {
+            self.priority_fee_tracker.get_priority_fee_estimates2(accounts, include_vote, lookback_period)
+        })
+    }
+}
+
+impl AtlasPriorityFeeEstimator {
+    pub fn new(
+        priority_fee_tracker: Arc<PriorityFeeTracker>,
+        rpc_url: String,
+        max_lookback_slots: usize,
+    ) -> Self {
+        let server = AtlasPriorityFeeEstimator {
+            priority_fee_tracker,
+            rpc_client: Some(RpcClient::new(rpc_url)),
+            max_lookback_slots,
+        };
+        server
+    }
+
+    fn execute_priority_fee_estimate_coordinator<F>(
```

Comment:
moved most of the logic - the actual call to fee tracker to a separate method and passing the function now so we could have two entry points to try different calls for comparison